lib/providers/trait_detail_provider.dart

```diff
@@ -22,15 +22,13 @@ class TraitDetailProvider extends ChangeNotifier {
     fetchData(getDetails: getDetails);
   }
 
-  Future<Device?> fetchData(
+  Future<void> fetchData(
```

Comment:
Please ensure if this was used just as others as well. Looks like changeNotifier is not called here as well.